Fix extraction issues

abhishekvm opened the pull request:
----
CRAW-208: http://www.nydailynews.com/newswires/sports/kershaw-wins-bellinger-hits-2-homers-dodgers-top-indians-article-1.3245615

Before:
"author": "",
"author_description": "",

Now:
"author": "The Associated Press",
"author_description": "THE ASSOCIATED PRESS",

----
CRAW-203: https://www.ft.com/content/d0a5c460-5044-11e7-a1f2-db19572361bb

Before:
"author":"",
"author_description":"",

Now:
"author": "Michael Power"
"author_description": "by: Michael Power, Investec"

----
CRAW-211: http://sdn.cioreview.com/cxoinsight/sdn-do-you-really-need-it-nid-24422-cid-147.html

Before:
"author": "CIO Review"
"author_description": "CIO Review"

Now:
"author": "Patrick Hubbard"
"author_description": "Patrick Hubbard, Head Geek, SolarWinds"

----
CRAW-212: http://www.philly.com/archive/john_timpane/Princetons-Tracy-K-Smith-named-US-Poet-Laureate.html

Fixed extra content extraction by adding rule in NODES_TO_REMOVE_PER_DOMAIN

----
CRAW-210: http://nation.foxnews.com/2017/06/13/sessions-inquisition-comes-empty-trump-unscathed-after-attorney-generals-testimony

Fixed extra content extraction by adding rule in NODES_TO_REMOVE_PER_DOMAIN

Before:
"author": ""
"author_description": ""

Now:
"author": "Hans A. von Spakovsky | Fox Opinion"
"author_description": "Hans A. von Spakovsky | Fox Opinion"

Commit:
Fix extraction issues

----
CRAW-208: http://www.nydailynews.com/newswires/sports/kershaw-wins-bellinger-hits-2-homers-dodgers-top-indians-article-1.3245615

Before:
"author": "",
"author_description": "",

Now:
"author": "The Associated Press",
"author_description": "THE ASSOCIATED PRESS",

----
CRAW-203: https://www.ft.com/content/d0a5c460-5044-11e7-a1f2-db19572361bb

Before:
"author":"",
"author_description":"",

Now:
"author": "Michael Power"
"author_description": "by: Michael Power, Investec"

----
CRAW-211: http://sdn.cioreview.com/cxoinsight/sdn-do-you-really-need-it-nid-24422-cid-147.html

Before:
"author": "CIO Review"
"author_description": "CIO Review"

Now:
"author": "Patrick Hubbard"
"author_description": "Patrick Hubbard, Head Geek, SolarWinds"

----
CRAW-212: http://www.philly.com/archive/john_timpane/Princetons-Tracy-K-Smith-named-US-Poet-Laureate.html

Fixed extra content extraction by adding rule in NODES_TO_REMOVE_PER_DOMAIN

----
CRAW-210: http://nation.foxnews.com/2017/06/13/sessions-inquisition-comes-empty-trump-unscathed-after-attorney-generals-testimony

Fixed extra content extraction by adding rule in NODES_TO_REMOVE_PER_DOMAIN

Before:
"author": ""
"author_description": ""

Now:
"author": "Hans A. von Spakovsky | Fox Opinion"
"author_description": "Hans A. von Spakovsky | Fox Opinion"

andresp99999 commented on src/main/java/de/jetwick/snacktory/ArticleTextExtractor.java:
```diff
@@ -2481,6 +2507,12 @@ protected String extractAuthorDescription(Document doc, String authorName){
+        // http://www.nydailynews.com/newswires/sports/kershaw-wins-bellinger-hits-2-homers-dodgers-top-indians-article-1.3245615
+        matches = doc.select("div[class=ra-credits]");
+        if (matches == null || matches.size() > 0){
+            return SHelper.innerTrim(matches.first().ownText());
```
Maybe add a DEBUG_AUTHOR_EXTRACTION log here?

Commit:
Addressed review comments

skyshard commented on src/main/java/de/jetwick/snacktory/ArticleTextExtractor.java:
```diff
@@ -2203,6 +2217,18 @@ protected String extractAuthorName(Document doc) {
+
+                    // http://nation.foxnews.com/2017/06/13/sessions-inquisition-comes-empty-trump-unscathed-after-attorney-generals-testimony
+                    if (matches == null || matches.size() == 0){
+                        matches = doc.select("p > strong:contains(By)");
```
This looks like it would be a slow selector

abhishekvm replied:
@skyshard  The article uses pretty much plain html tags without distinguishing css.

abhishekvm replied:
@skyshard - Can you confirm what can be done here? Should we merge this as is or remove fix for nation.foxnews.com?

andresp99999 replied:
@abhishek-airpr let's remove the nation.foxnews.com hack for now, after that feel free to merge it.

Commit:
Merge branch 'master' of https://github.com/skyshard/snacktory into abhishek/fix_author_content_extraction

Commit:
Merge branch 'master' of https://github.com/skyshard/snacktory into abhishek/fix_author_content_extraction

Commit:
Fixed Author Extraction
----
CRAW-223: Added a Regex match as a last check if author name is not extracted
http://www.reuters.com/article/us-mexico-oil-ninth-idUSKBN19A2M9
Author: Adriana Barrerau
Author Desc: (Reporting by Adriana Barrera)

http://www.reuters.com/article/us-safrica-mining-idUSKBN19A2PY
Author: James Macharia
Author Desc: James Macharia

----
CRAW-229: upi.com
http://www.upi.com/Entertainment_News/TV/2017/06/19/Star-Trek-Discovery-gets-a-premiere-date-Sept-24/8671497893578/
Author: Karen Butler
Author Desc: Karen Butler

http://www.upi.com/Defense-News/2017/06/19/King-Aerospace-recieves-EO-5-aircraft-contract/2311497885914/
Author: Stephen Carlson
Author Desc: By Stephen Carlson | June 19, 2017 at 1:48 PM Follow @upi

----
CRAW-230: foxnews.com
http://www.foxnews.com/tech/2017/06/19/southwest-border-cities-to-use-iris-scanning-technology-to-identify-inmates-illegal-immigrants.html
Author: Ray Bogan
Author Desc: Ray Bogan is a Fox News multimedia reporter based in El Paso, Texas. Follow him on twitter:

http://www.foxnews.com/politics/2017/06/19/special-counsel-muellers-legal-team-continues-to-raise-questions.html
Author: Doug McKelway
Author Desc: Doug McKelway joined Fox News Channel (FNC) in November 2010 and serves as a Washington-based correspondent. Click here for more information on Doug McKelway.

----
CRAW-222: uk.finance.yahoo.com
https://uk.finance.yahoo.com/news/trump-likely-won-apos-t-212100132.html
Author: Jacob Pramuk
Author Desc: http://www.cnbc.com

----
CRAW-225: http://www.stltoday.com/news/national/coke-tweet-targeting-pittsburghers-uses-map-of-philadelphia/article_df15f84f-bac7-5ef6-890c-28b966fa5269.html
- Fixed boundaries of article text

----
CRAW-227: http://www.cbc.ca/news/canada/calgary/sled-island-preview-1.4167762
- Fixed boundaries of article text

andresp99999 committed:
Merge pull request #45 from skyshard/abhishek/fix_author_extraction_issues

Fixed Author Extraction

Commit:
Merge branch 'abhishek/fix_author_content_extraction' of https://github.com/skyshard/snacktory into abhishek/fix_author_content_extraction

Commit:
Removed fix for nation.foxnews.com which require very site specific hack.
